src/transformers/models/whisper/tokenization_whisper.py

```diff
@@ -988,3 +992,12 @@ def _find_longest_common_sequence(sequences):
     total_sequence.extend(left_sequence)
 
     return total_sequence
+
+
+def _strip_prompt(token_ids: List[int], prompt_token_id: int, all_special_ids: List[int]):
+    has_prompt = isinstance(token_ids, list) and token_ids and token_ids[0] == prompt_token_id
+    if has_prompt:
+        for i in range(1, len(token_ids)):
+            if token_ids[i] in all_special_ids:
```

Comment:
So we just clip to the first time we hit `<|startoftranscript|>`? Think if we know that `has_prompt` is True then we can return the token ids from `<|startoftranscript|>` onwards explicitly (think this will make the code easier to understand)

Reply:
Much simpler! It's now short enough I'm wondering if you'd prefer including the below directly in the `_decode` instead of in the static method
```py
if skip_special_tokens:
            decoder_start_token_id = self.convert_tokens_to_ids("<|startoftranscript|>")
            decoder_start_token_id_idx = token_ids.index(decoder_start_token_id)
            token_ids = token_ids[decoder_start_token_id_idx:] # strip prompt if it exists
```